datafusion/physical-plan/src/windows/bounded_window_agg_exec.rs

```diff
@@ -257,17 +257,11 @@ impl ExecutionPlan for BoundedWindowAggExec {
     fn required_input_ordering(&self) -> Vec<Option<LexRequirement>> {
         let partition_bys = self.window_expr()[0].partition_by();
         let order_keys = self.window_expr()[0].order_by();
-        if self.input_order_mode != InputOrderMode::Sorted
-            || self.ordered_partition_by_indices.len() >= partition_bys.len()
-        {
-            let partition_bys = self
-                .ordered_partition_by_indices
-                .iter()
-                .map(|idx| &partition_bys[*idx]);
-            vec![calc_requirements(partition_bys, order_keys)]
-        } else {
-            vec![calc_requirements(partition_bys, order_keys)]
```

Comment:
when `self.input_order_mode` is `InputOrderMode::Sorted`, it is guaranteed that `self.ordered_partition_by_indices.len() == partition_bys.len()`. Hence, `else` branch was never being triggerred in the previous version. I think, that check is redundant.